components/org.wso2.carbon.identity.password.expiry/src/main/java/org/wso2/carbon/identity/password/expiry/models/PasswordExpiryRule.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.password.expiry.models;
+
+import org.apache.commons.lang.StringUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Class to represent a password expiry rule.
+ */
+public class PasswordExpiryRule {
+
+    private int priority;
+    private int expiryDays;
+    private PasswordExpiryRuleAttributeEnum attribute;
+    private PasswordExpiryRuleOperatorEnum operator;
+    private List<String> values;
+    private static final String RULE_SPLIT_REGEX = ",(?=(?:[^']*'[^']*')*[^']*$)";
+
+    public PasswordExpiryRule(int priority, int expiryDays, PasswordExpiryRuleAttributeEnum attribute,
+                              PasswordExpiryRuleOperatorEnum operator, List<String> values) {
+
+        this.priority = priority;
+        this.expiryDays = expiryDays;
+        this.attribute = attribute;
+        this.operator = operator;
+        this.values = values;
+    }
+
+    public PasswordExpiryRule(String rule) throws IllegalArgumentException{
+
+        try {
+            // Rule format: "priority,expiryDays,attribute,operator,value1,value2, ...".
+            // Eg: "1,40,roles,eq,12ec01e1-aa45-8a485d10c8fa,cc40ad49-8435-75fa1b627332".
+            String[] ruleSections = rule.split(RULE_SPLIT_REGEX);
+            if (ruleSections.length < 4) {
+                throw new IllegalArgumentException("Invalid rule format: not enough parts in the rule definition.");
+            }
+
+            this.priority = Integer.parseInt(ruleSections[0].trim());
+            this.expiryDays = Integer.parseInt(ruleSections[1].trim());
+            this.attribute = PasswordExpiryRuleAttributeEnum.fromString(ruleSections[2].trim());
+            this.operator = PasswordExpiryRuleOperatorEnum.fromString(ruleSections[3].trim());
+            this.values = new ArrayList<>();
+
+            // Extract values from the rule removing quotes if present.
+            for (int i = 4; i < ruleSections.length; i++) {
+                String value = ruleSections[i].trim();
+                if ((StringUtils.startsWith(value, "'") && StringUtils.endsWith(value, "'")) ||
```

Comment:
Let's add a comment with an example case what type of value going to process. As mentioned earlier there can be roles/groups with single quotes or double quotes.

Reply:
Added